app/graphql/resolvers/rdf.ts

```diff
@@ -277,20 +277,13 @@ export const dataCubeComponents: NonNullable<
 
 export const dataCubeMetadata: NonNullable<QueryResolvers["dataCubeMetadata"]> =
   async (_, { locale, cubeFilter }, { setup }, info) => {
-    const { loaders, sparqlClient } = await setup(info);
+    const { sparqlClient } = await setup(info);
     const { iri, latest = true } = cubeFilter;
-    const rawCube = await loaders.cube.load(iri);
-
-    if (!rawCube) {
-      throw new Error("Cube not found!");
-    }
-
-    const cube = latest ? await getLatestCube(rawCube) : rawCube;
+    const cube = await new LightCube({ iri, locale, sparqlClient }).promote(
+      !!latest
+    );
```

Comment:
Nit: I read `promote` as like a "write" on the cube,  instead of a "read", I wonder if this way would not be clearer ?

```
const cube = await new LightCube().fetch({ latest: !!latest })
```

Reply:
Maybe "init" would also work? Here in fact we are not fetching any information for the cube, but rather (potentially) return a new, latest instance of `LightCube`. As we have `fetchMetadata`, and will have other methods soon, I wonder if "fetch" wouldn't feel like a combined fetch of everything related to the cube?